src/screens/augmentedReality/ARShowScreen.js

```diff
@@ -174,38 +176,52 @@ const AugmentedRealityView = ({ sceneNavigator }) => {
 
   // TODO: these data can be updated according to the data coming from the server when the
   //       real 3D models arrive
-  const position = [0, -1, -5];
+  const position = [0, 0, 0];
   const rotation = [0, 0, 0];
-  const scale = [0.02, 0.02, 0.02];
+  const scale = [0.002, 0.002, 0.002];
+
+  ViroARTrackingTargets.createTargets({
+    targetImage: {
+      physicalWidth: 0.2, // real world width in meters
+      source: { uri: object.target },
+      type: 'image'
+    }
+  });
 
   return (
     <ViroARScene dragType="FixedToWorld">
       <ViroAmbientLight color={'#fff'} />
 
-      {!!object.mp3 && !isObjectLoading && (
-        <ViroSound
-          source={{ uri: object.mp3 }}
-          paused={!isStartAnimationAndSound}
-          onFinish={() => setIsStartAnimationAndSound(false)}
-        />
-      )}
+      <ViroARImageMarker
+        onAnchorFound={() => setIsStartAnimationAndSound(true)} // animation and sound file are started after the image is recognised
+        pauseUpdates={true} // prevents the model from continuous jumping
+        target={'targetImage'}
+      >
+        {!!object.mp3 && !isObjectLoading && (
+          <ViroSound
+            source={{ uri: object.mp3 }}
+            paused={!isStartAnimationAndSound}
+            onFinish={() => setIsStartAnimationAndSound(false)}
+          />
+        )}
 
-      <Viro3DObject
-        source={{ uri: object.vrx }}
-        resources={[{ uri: object.png }]}
-        type="VRX"
-        position={position}
-        rotation={rotation}
-        scale={scale}
-        onLoadStart={() => setIsObjectLoading(true)}
-        onLoadEnd={() => setIsObjectLoading(false)}
-        onError={() => alert(texts.augmentedReality.arShowScreen.objectLoadErrorAlert)}
-        animation={{
-          loop: true,
-          name: object.animationName,
-          run: isStartAnimationAndSound
-        }}
-      />
+        <Viro3DObject
+          source={{ uri: object.vrx }}
+          resources={[{ uri: object.png }]}
```

Comment:
what happens if `object.vrc` or `object.png` would be undefined? is this checked somewhere before?

Reply:
I will need to add a control in the parser for these. If undefined, the application crashes!

Reply:
Added a control and prevented the application from crashing.

https://github.com/ikuseiGmbH/smart-village-app-app/pull/592/commits/cb2f8b030d0be12176bb12acbbc265177dbd7e43